File: day1/day1.rb
def part1(input)
  input.split("\n\n").map { |a| a.split("\n").map(&:to_i).sum }.max
end

def part2(input)
  input.split("\n\n").map { |a| a.split("\n").map(&:to_i).sum }.sort.reverse[0..2].sum
end

test_input = <<-INPUT
1000
2000
3000

4000

5000
6000

7000
8000
9000

10000
INPUT
result = part1(test_input)
raise result.inspect if result != 24000

result = part2(test_input)
raise result.inspect if result != 45000

input = File.read("input.txt")
puts "part1 - #{part1(input)}"
puts "part2 - #{part2(input)}"

File: day1/day1.js
const fs = require('fs');

const sum = (arr) => arr.reduce((accumulator, value) => {
    return accumulator + value;
  }, 0);

const part1 = (input) => {
    return input
        .split("\n\n")
        .map(a => sum(a.split("\n").map(a => parseInt(a))))
        .sort((a, b) => (a - b))
        .reverse()[0]
}

const part2 = (input) => {
    const elves = input
        .split("\n\n")
        .map(a => sum(a.split("\n").map(a => parseInt(a))))
        .sort((a, b) => (a - b))
        .reverse()
        .slice(0, 3);
    return sum(elves);
}

const test_input = 
`1000
2000
3000

4000

5000
6000

7000
8000
9000

10000`;
let result = part1(test_input);
if (result != 24000) {
    throw(result)
}

result = part2(test_input);
if (result != 45000) {
    throw(result)
}

const input = fs.readFileSync("input.txt", "utf8");
console.log(`part1 - ${part1(input)}`);
console.log(`part2 - ${part2(input)}`);
